lib/note-content-editor.jsx

```diff
@@ -120,20 +123,20 @@ export default class NoteContentEditor extends React.Component {
 		this.setState( { editorState }, announceChanges );
 	}
 
-	componentWillReceiveProps( { content: newContent } ) {
-		const { content: oldContent } = this.props;
+	componentWillReceiveProps( { content: newContent, filter: nextFilter } ) {
+		const { content: oldContent, filter: prevFilter } = this.props;
 		const { editorState: oldEditorState } = this.state;
 
-		if ( newContent === oldContent ) {
-			return; // identical to previous `content` prop
-		}
-
-		if ( newContent === plainTextContent( oldEditorState ) ) {
+		if (
+			( newContent === plainTextContent( oldEditorState ) ) &&
+			( nextFilter === prevFilter )
+		) {
 			return; // identical to rendered content
 		}
 
 		let newEditorState = EditorState.createWithContent(
-			ContentState.createFromText( newContent, '\n' )
+			ContentState.createFromText( newContent, '\n' ),
```

Comment:
@nfcampos we already have a few issues with the cursor. thanks for pointing this out. I've heard some discussions lately about saving the cursor position on each note, which would definitely help here. I'll take a look and test with this.

when searching, I think it's possible that losing the cursor could be okay if we have to but I will try and rearrange so we don't